### ISSUE TITLE
UseInterpreter is set in any project referencing a class library that references MAUI

### ISSUE THREAD

**Issue by rolfbjarne** [dotnet/maui#21653]
### Description

The following situation:

* Create a `net8.0` class library that references the `Microsoft.Maui.Controls` NuGet.
* Create an executable project that references the class library.
* The `UseInterpreter` property is now set to `true` in the executable project.

### Steps to Reproduce

If the executable project is a `macOS` project (`net8.0-macos`), then the build fails (https://github.com/xamarin/xamarin-macios/issues/20398 - which we'll fix on our end, but the fact still remains that MAUI sets `UseInterpreter` for all platforms).

### Link to public reproduction project repository

https://github.com/rolfbjarne/mauitests/tree/main/maui-classlib-macos

### Version with bug

8.0.10 SR3

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Relevant log output

```shell
Binlog: https://github.com/rolfbjarne/mauitests/blob/main/maui-classlib-macos/macos/msbuild.binlog
```

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio 17.6.11(build400)(8.0.20&8.0.14&8.0.10). Can repro on macOS platform with sample project. 
![image](https://github.com/dotnet/maui/assets/77434865/abd01aba-f8b2-4cc3-b329-13ad5f225e4d)

**Comment by PureWeen** [comment]
@rolfbjarne is this something we still want to act on for .NET10 ?

**Comment by rolfbjarne** [comment]
@PureWeen yes, I think so. This was worked around on our end by ignoring "UseInterpreter" when building with CoreCLR (because CoreCLR doesn't have an interpreter), but an interpreter for CoreCLR is being worked on, so the workaround might not work in the future.
